react/FormSubmit.tsx

```diff
@@ -33,29 +35,35 @@ const messages = defineMessages({
 })
 
 export default function FormSubmit({
-  label = messages.submitButton.id,
-}: FormSubmitProps) {
-  const { loading, userInputError, serverError } = useContext(SubmitContext)
+                                     label = messages.submitButton.id,
+                                     labelId = ''
+                                   }: FormSubmitProps) {
+  const {loading, userInputError, serverError} = useContext(SubmitContext)
   const handles = useCssHandles(CSS_HANDLES)
 
   return (
     <div className={handles.formSubmitContainer}>
       <div className={handles.formSubmitButton}>
         <Button type="submit" isLoading={loading}>
-          <IOMessage id={label} />
+          {labelId ? (
+            getMessage(`store/form.label-${labelId}`)
```

Comment:
I prefer the labelId to be the whole key, i.e.: `store/form.label-whatever`, preventing misunderstandings. And if ain't wrong, this is the pattern, to receive the whole message key, instead of part.